### PR TITLE
Make tests name-independent

### PR DESCRIPTION
**Pull request by martinmaly**
Tests no longer rely on specific package names generated by the server to
allow the server to change the name creation algorithm.

### CODE REVIEW

File: porch/apiserver/pkg/registry/porch/packagerevision.go [hunk 1]
```diff
@@ -98,23 +98,17 @@ func (r *packageRevisions) Create(ctx context.Context, runtimeObject runtime.Obj
...
 	}
 
-	if name != expectedName {
-		return nil, apierrors.NewBadRequest(fmt.Sprintf("name should be %q", expectedName))
-	}
-
-	nameTokens, err := ParseName(name)
-	if err != nil {
-		return nil, apierrors.NewBadRequest(fmt.Sprintf("invalid name %q", name))
+	repositoryName := obj.Spec.RepositoryName
```
Comment (justinsb) [inline porch/apiserver/pkg/registry/porch/packagerevision.go:105]: Should we check that spec.repositoryName matches the value encoded name?  Or is the plan to just move away that restriction entirely?

File: porch/repository/pkg/git/git_test.go [hunk 2]
```diff
@@ -521,29 +526,33 @@ func TestListPackagesDrafts(t *testing.T) {
...
 	}
 
-	got := map[string]v1alpha1.PackageRevisionLifecycle{}
+	got := map[repository.PackageRevisionName]v1alpha1.PackageRevisionLifecycle{}
 	for _, r := range revisions {
 		rev, err := r.GetPackageRevision()
 		if err != nil {
 			t.Errorf("GetPackageRevision failed for %q: %v", r.Name(), err)
 		}
-		got[r.Name()] = rev.Spec.Lifecycle
```
Comment (justinsb) [inline porch/repository/pkg/git/git_test.go:546]: If we do this a lot, we could define a method `Key()` which returns 

```
repository.PackageRevisionName{
			Repository: rev.Spec.RepositoryName,
			Package:    rev.Spec.PackageName,
			Revision:   rev.Spec.Revision,
		}
```

File: porch/repository/pkg/git/git_test.go [hunk 3]
```diff
@@ -647,7 +660,11 @@ func TestDeletePackages(t *testing.T) {
 	for len(all) > 0 {
 		// Delete one of the packages
 		deleting := all[0]
-		name := deleting.Name()
+		pr, err := deleting.GetPackageRevision()
+		if err != nil {
+			t.Fatalf("GetPackageRevision of %q faile: %v", deleting.Name(), err)
```
Comment (justinsb) [inline porch/repository/pkg/git/git_test.go:665]: Supernit: failed

File: porch/repository/pkg/git/testing_repo.go [hunk 4]
```diff
@@ -183,37 +183,45 @@ func findFile(t *testing.T, tree *object.Tree, path string) *object.File {
...
 
-func findPackage(t *testing.T, revisions []repository.PackageRevision, name string) repository.PackageRevision {
+func findPackage(t *testing.T, revisions []repository.PackageRevision, name repo.PackageRevisionName) repository.PackageRevision {
 	for _, r := range revisions {
-		if r.Name() == name {
+		rev, err := r.GetPackageRevision()
+		if err != nil {
+			t.Fatalf("GetPackageRevision of %q failed: %v", r.Name(), err)
+		}
+		if rev.Spec.RepositoryName == name.Repository &&
```
Comment (justinsb) [inline porch/repository/pkg/git/testing_repo.go:192]: An advantage of `Key()` could be that I think we could then just do a `==` check against the two structs.

File: porch/apiserver/pkg/registry/porch/packagerevision.go (hunk 1 above)
Comment (justinsb) [reply porch/apiserver/pkg/registry/porch/packagerevision.go:105]: For Create, I think we expect it normally to be called with a Name, and I don't think we should default it if not set - I think that would be surprising and would likely break validation.  The functionality is available in GenerateName if we do want it, though it isn't used very often AFAIK

File: porch/apiserver/pkg/e2e/suite.go [hunk 5]
```diff
@@ -637,3 +638,16 @@ func normalizeYamlOrdering(t *testing.T, contents string) string {
 	}
 	return stable.String()
 }
+
+func FindPackageRevision(t *testing.T, packages *porchapi.PackageRevisionList, name repository.PackageRevisionName) *porchapi.PackageRevision {
```
[the remark this answers is not in the document]
Comment (martinmaly) [reply porch/apiserver/pkg/e2e/suite.go:642]: done

File: porch/repository/pkg/git/git_test.go (hunk 3 above)
Comment (martinmaly) [reply porch/repository/pkg/git/git_test.go:665]: thanks!

**Review by justinsb — APPROVED** [review]

File: porch/repository/pkg/git/testing_repo.go (hunk 4 above)
Comment (martinmaly) [reply porch/repository/pkg/git/testing_repo.go]: done

File: porch/apiserver/pkg/registry/porch/packagerevision.go (hunk 1 above)
Comment (martinmaly) [reply porch/apiserver/pkg/registry/porch/packagerevision.go:105]: will add TODO with the 2nd PR

**Pull request merged**